### ISSUE TITLE
Zipkin header promotion not behaving as documented

### ISSUE THREAD

**Issue by alext** [cloudfoundry/routing-release#110]
## Issue

I'm not seeing the Zipkin headers being promoted as documented here: https://docs.cloudfoundry.org/concepts/http-routing.html#-http-headers-for-zipkin-tracing.  Instead, GoRouter looks to be passing them through unmodified if they're present in the request.

## Context

It looks as though this behaviour was changed in https://github.com/cloudfoundry/gorouter/commit/fa255ae, but there's no explaination there as to why this change was made.

## Steps to Reproduce

Push the following app:
```Go
package main

import (
	"log"
	"net/http"
	"net/http/httputil"
	"os"
)

func main() {
	http.HandleFunc("/", func(w http.ResponseWriter, req *http.Request) {
		info, err := httputil.DumpRequest(req, false)
		if err != nil {
			http.Error(w, err.Error(), http.StatusInternalServerError)
			return
		}
		w.Write(info)
	})
	log.Fatal(http.ListenAndServe(":"+os.Getenv("PORT"), nil))
}
```
Curl the app:
`curl -H 'X-B3-Spanid: 1234' -H 'X-B3-Traceid: 1234' http://app.example.com/`

[ordered list the process to finding and recreating the issue, example below]

## Expected result

I expect `X-B3-ParentSpanid` to be set to `1234` and `X-B3-Spanid` to be set to a new value.

## Current result

`X-B3-Spanid` and `X-B3-TraceId` are passed through unmodified, and no `X-B3ParentSpanid` header is added.

## Possible Fix

If this is the intended behaviour, can the documentation be updated to reflect this.

**Comment by shalako** [comment]
Current behavior is based on guidance from @adriancole

**Comment by adriancole** [comment]
right. since the router is not sending data to zipkin, it should not change
the headers (like creating a child). It acts like a pass through which is
only affecting the choice of the root ids.

**Comment by adriancole** [comment]
ps the rationale for not changing incoming ids is that the parent child link becomes lost when this is done and data isn't sent to the tracing system. A "lost parent" makes it impossible to reconstruct the trace tree accurately. Since the router doesn't send data out, the safe mode is to propagate what it received. I think in Amazon Xray, they call this "pass through mode" or similar

**Comment by shalako** [comment]
Thank you, @adriancole

**Comment by shubhaat** [comment]
@alext - as you have identified correctly our docs are outdated, and we will update them to reflect current behavior. We will keep the story open until we fix the issue. Thank you!

**Comment by shalako** [comment]
Someone else asked about this in https://github.com/cloudfoundry/gorouter/issues/221

**Comment by jblackman** [comment]
Just to add my 10c (as the "Someone else" above) our use case is to easily be able to determine the amount of load that one app is putting on another. Having parent span id propagation works very well for this.

**Comment by sajanep** [comment]
Hello,
If this is intended, what is the recommendation here. As a consumer of Zipkin in CF(owner for a number of services), do I need to generate a new value for x-b3-span-id and adjust x-b3-parentid accordingly with each service call. Sorry. I am a newbie to Zipkin.

**Comment by shalako** [comment]
Closing this, as current behavior is intentional. Docs PR is here: https://github.com/cloudfoundry/docs-cloudfoundry-concepts/pull/96 (thank you, @alext !). 

This change was released in routing-release 0.149.0: https://github.com/cloudfoundry/routing-release/releases/tag/0.149.0
